diff -bcrN pcre-8.01.org/config.sub pcre-8.01/config.sub
*** pcre-8.01.org/config.sub	2010-01-19 23:03:14.000000000 +0800
--- pcre-8.01/config.sub	2010-03-03 14:01:54.226317943 +0800
***************
*** 1300,1306 ****
  	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
  	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
  	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
! 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -irx*)
  	# Remember, each alternative MUST END IN *, to match a version number.
  		;;
  	-qnx*)
--- 1300,1306 ----
  	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
  	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
  	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
! 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -irx* | -es*)
  	# Remember, each alternative MUST END IN *, to match a version number.
  		;;
  	-qnx*)
diff -bcrN pcre-8.01.org/pcretest.c pcre-8.01/pcretest.c
*** pcre-8.01.org/pcretest.c	2009-10-05 18:40:30.000000000 +0800
--- pcre-8.01/pcretest.c	2010-03-03 14:03:14.110330660 +0800
***************
*** 909,915 ****
        ((stack_size = get_value((unsigned char *)argv[op+1], &endptr)),
          *endptr == 0))
      {
! #if defined(_WIN32) || defined(WIN32)
      printf("PCRE: -S not supported on this OS\n");
      exit(1);
  #else
--- 909,915 ----
        ((stack_size = get_value((unsigned char *)argv[op+1], &endptr)),
          *endptr == 0))
      {
! #if defined(_WIN32) || defined(WIN32) || defined(__es__)
      printf("PCRE: -S not supported on this OS\n");
      exit(1);
  #else
