diff -burN binutils-2.19.org/bfd/config.bfd binutils-2.19/bfd/config.bfd
--- binutils-2.19.org/bfd/config.bfd	2008-04-17 03:02:01.000000000 +0900
+++ binutils-2.19/bfd/config.bfd	2009-01-10 18:20:26.000000000 +0900
@@ -526,6 +526,10 @@
     targ_defvec=bfd_elf32_i386_vec
     targ_selvecs="i386coff_vec i386aout_vec"
     ;;
+  i[3-7]86-*-es*)
+    targ_defvec=bfd_elf32_i386_vec
+    targ_selvecs="i386coff_vec i386aout_vec"
+    ;;
   i[3-7]86-*-darwin* | i[3-7]86-*-macos10* | i[3-7]86-*-rhapsody*)
     targ_defvec=mach_o_le_vec
     targ_selvecs="mach_o_le_vec mach_o_be_vec mach_o_fat_vec pef_vec pef_xlib_vec sym_vec"
diff -burN binutils-2.19.org/config.sub binutils-2.19/config.sub
--- binutils-2.19.org/config.sub	2008-04-14 18:28:35.000000000 +0900
+++ binutils-2.19/config.sub	2009-01-10 18:21:21.000000000 +0900
@@ -1266,7 +1266,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -burN binutils-2.19.org/configure binutils-2.19/configure
--- binutils-2.19.org/configure	2008-09-04 11:18:16.000000000 +0900
+++ binutils-2.19/configure	2009-01-10 18:39:30.000000000 +0900
@@ -1881,7 +1881,7 @@
 # binutils, gas and ld appear in that order because it makes sense to run
 # "make check" in that particular order.
 # If --enable-gold is used, "gold" will replace "ld".
-host_tools="texinfo byacc flex bison binutils gas ld fixincludes gcc sid sim gdb make patch prms send-pr gprof etc expect dejagnu ash bash bzip2 m4 autoconf automake libtool diff rcs fileutils shellutils time textutils wdiff find uudecode hello tar gzip indent recode release sed utils guile perl gawk findutils gettext zip fastjar gnattools"
+host_tools="byacc flex bison binutils gas ld fixincludes gcc sid sim gdb make patch prms send-pr gprof etc expect dejagnu ash bash bzip2 m4 autoconf automake libtool diff rcs fileutils shellutils time textutils wdiff find uudecode hello tar gzip indent recode release sed utils guile perl gawk findutils gettext zip fastjar gnattools"
 
 # libgcj represents the runtime libraries only used by gcj.
 libgcj="target-libffi \
@@ -2251,6 +2251,9 @@
   *-*-rtems*)
     noconfigdirs="$noconfigdirs target-libgloss ${libgcj}"
     ;;
+  *-*-es*)
+    noconfigdirs="$noconfigdirs ${libgcj}"
+    ;;
     # The tpf target doesn't support gdb yet.
   *-*-tpf*)
     noconfigdirs="$noconfigdirs target-newlib target-libgloss target-libiberty ${libgcj} target-libmudflap gdb tcl tk libgui itcl"
diff -burN binutils-2.19.org/configure.ac binutils-2.19/configure.ac
--- binutils-2.19.org/configure.ac	2008-10-16 22:45:42.000000000 +0900
+++ binutils-2.19/configure.ac	2009-01-10 18:22:37.000000000 +0900
@@ -500,6 +500,9 @@
   *-*-rtems*)
     noconfigdirs="$noconfigdirs target-libgloss ${libgcj}"
     ;;
+  *-*-es*)
+    noconfigdirs="$noconfigdirs ${libgcj}"
+    ;;
     # The tpf target doesn't support gdb yet.
   *-*-tpf*)
     noconfigdirs="$noconfigdirs target-newlib target-libgloss target-libiberty ${libgcj} target-libmudflap gdb tcl tk libgui itcl"
diff -burN binutils-2.19.org/gas/configure.tgt binutils-2.19/gas/configure.tgt
--- binutils-2.19.org/gas/configure.tgt	2008-10-02 17:07:17.000000000 +0900
+++ binutils-2.19/gas/configure.tgt	2009-01-10 18:23:28.000000000 +0900
@@ -213,6 +213,7 @@
   i386-*-msdosdjgpp* \
   | i386-*-go32*)			fmt=coff em=go32 ;;
   i386-*-rtems*)			fmt=elf ;;
+  i386-*-es*)				fmt=elf ;;
   i386-*-gnu*)				fmt=elf em=gnu ;;
   i386-*-mach*)				fmt=aout em=mach ;;
   i386-*-msdos*)			fmt=aout ;;
diff -burN binutils-2.19.org/ld/configure.tgt binutils-2.19/ld/configure.tgt
--- binutils-2.19.org/ld/configure.tgt	2008-08-09 14:35:13.000000000 +0900
+++ binutils-2.19/ld/configure.tgt	2009-01-10 18:24:55.000000000 +0900
@@ -181,6 +181,7 @@
 i[3-7]86-*-lynxos*)	targ_emul=i386lynx ;;
 i[3-7]86-*-coff)	targ_emul=i386coff ;;
 i[3-7]86-*-rtems*)	targ_emul=elf_i386 ;;
+i[3-7]86-*-es*)		targ_emul=elf_i386 ;;
 i[3-7]86-*-aros*)	targ_emul=elf_i386 ;;
 i[3-7]86-*-rdos*)	targ_emul=elf_i386 ;;
 i[3-7]86-*-bsd)		targ_emul=i386bsd ;;
