diff -bcrN libxml2-2.7.3.org/config.sub libxml2-2.7.3/config.sub
*** libxml2-2.7.3.org/config.sub	2009-06-29 12:11:01.920856443 +0800
--- libxml2-2.7.3/config.sub	2009-06-30 00:47:21.828645092 +0800
***************
*** 1217,1223 ****
  	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
  	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
  	      | -chorusos* | -chorusrdb* \
! 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
  	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
  	      | -uxpv* | -beos* | -mpeix* | -udk* \
  	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
--- 1217,1223 ----
  	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
  	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
  	      | -chorusos* | -chorusrdb* \
! 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* | -es* \
  	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
  	      | -uxpv* | -beos* | -mpeix* | -udk* \
  	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
diff -bcrN libxml2-2.7.3.org/Makefile.am libxml2-2.7.3/Makefile.am
*** libxml2-2.7.3.org/Makefile.am	2009-06-29 12:11:01.812856581 +0800
--- libxml2-2.7.3/Makefile.am	2009-06-30 00:47:30.512645481 +0800
***************
*** 6,15 ****
  
  INCLUDES = -I$(top_builddir)/include -I@srcdir@/include @THREAD_CFLAGS@ @Z_CFLAGS@ 
  
! noinst_PROGRAMS=testSchemas testRelax testSAX testHTML testXPath testURI \
!                 testThreads testC14N testAutomata testRegexp \
!                 testReader testapi testModule runtest runsuite testchar \
! 		testdict runxmlconf testrecurse
  
  bin_PROGRAMS = xmllint xmlcatalog
  
--- 6,15 ----
  
  INCLUDES = -I$(top_builddir)/include -I@srcdir@/include @THREAD_CFLAGS@ @Z_CFLAGS@ 
  
! #noinst_PROGRAMS=testSchemas testRelax testSAX testHTML testXPath testURI \
! #                testThreads testC14N testAutomata testRegexp \
! #                testReader testapi testModule runtest runsuite testchar \
! #		testdict runxmlconf testrecurse
  
  bin_PROGRAMS = xmllint xmlcatalog
  
***************
*** 51,185 ****
  m4datadir = $(datadir)/aclocal
  m4data_DATA = libxml.m4
  
! runtest_SOURCES=runtest.c
! runtest_LDFLAGS = 
! runtest_DEPENDENCIES = $(DEPS)
! runtest_LDADD= @BASE_THREAD_LIBS@ @RDL_LIBS@ $(LDADDS)
! 
! testrecurse_SOURCES=testrecurse.c
! testrecurse_LDFLAGS = 
! testrecurse_DEPENDENCIES = $(DEPS)
! testrecurse_LDADD= @BASE_THREAD_LIBS@ @RDL_LIBS@ $(LDADDS)
! 
! testchar_SOURCES=testchar.c
! testchar_LDFLAGS = 
! testchar_DEPENDENCIES = $(DEPS)
! testchar_LDADD= @RDL_LIBS@ $(LDADDS)
! 
! testdict_SOURCES=testdict.c
! testdict_LDFLAGS = 
! testdict_DEPENDENCIES = $(DEPS)
! testdict_LDADD= @RDL_LIBS@ $(LDADDS)
! 
! runsuite_SOURCES=runsuite.c
! runsuite_LDFLAGS = 
! runsuite_DEPENDENCIES = $(DEPS)
! runsuite_LDADD= @RDL_LIBS@ $(LDADDS)
  
  xmllint_SOURCES=xmllint.c
  xmllint_LDFLAGS = 
  xmllint_DEPENDENCIES = $(DEPS)
  xmllint_LDADD=  @RDL_LIBS@ $(LDADDS)
  
! testSAX_SOURCES=testSAX.c
! testSAX_LDFLAGS = 
! testSAX_DEPENDENCIES = $(DEPS)
! testSAX_LDADD= $(LDADDS)
! 
! testHTML_SOURCES=testHTML.c
! testHTML_LDFLAGS = 
! testHTML_DEPENDENCIES = $(DEPS)
! testHTML_LDADD= $(LDADDS)
  
  xmlcatalog_SOURCES=xmlcatalog.c
  xmlcatalog_LDFLAGS = 
  xmlcatalog_DEPENDENCIES = $(DEPS)
  xmlcatalog_LDADD= @RDL_LIBS@ $(LDADDS)
  
! testXPath_SOURCES=testXPath.c
! testXPath_LDFLAGS = 
! testXPath_DEPENDENCIES = $(DEPS)
! testXPath_LDADD= $(LDADDS)
! 
! testC14N_SOURCES=testC14N.c
! testC14N_LDFLAGS = 
! testC14N_DEPENDENCIES = $(DEPS)
! testC14N_LDADD= $(LDADDS)
! 
! testThreads_SOURCES=testThreads@THREADS_W32@.c
! testThreads_LDFLAGS = 
! testThreads_DEPENDENCIES = $(DEPS)
! testThreads_LDADD= @BASE_THREAD_LIBS@  $(LDADDS)
! 
! testURI_SOURCES=testURI.c
! testURI_LDFLAGS = 
! testURI_DEPENDENCIES = $(DEPS)
! testURI_LDADD= $(LDADDS)
! 
! testRegexp_SOURCES=testRegexp.c
! testRegexp_LDFLAGS = 
! testRegexp_DEPENDENCIES = $(DEPS)
! testRegexp_LDADD= $(LDADDS)
! 
! testAutomata_SOURCES=testAutomata.c
! testAutomata_LDFLAGS = 
! testAutomata_DEPENDENCIES = $(DEPS)
! testAutomata_LDADD= $(LDADDS)
! 
! testSchemas_SOURCES=testSchemas.c
! testSchemas_LDFLAGS = 
! testSchemas_DEPENDENCIES = $(DEPS)
! testSchemas_LDADD= $(LDADDS)
! 
! testRelax_SOURCES=testRelax.c
! testRelax_LDFLAGS = 
! testRelax_DEPENDENCIES = $(DEPS)
! testRelax_LDADD= $(LDADDS)
! 
! testReader_SOURCES=testReader.c
! testReader_LDFLAGS = 
! testReader_DEPENDENCIES = $(DEPS)
! testReader_LDADD= $(LDADDS)
! 
! testModule_SOURCES=testModule.c
! testModule_LDFLAGS = 
! testModule_DEPENDENCIES = $(DEPS)
! testModule_LDADD= $(LDADDS)
! 
! noinst_LTLIBRARIES = testdso.la
! testdso_la_SOURCES = testdso.c
! testdso_la_LDFLAGS = -module -rpath $(libdir)
  
  # that one forces the rebuild when "make rebuild" is run on doc/
! rebuild_testapi:
! 	-@(if [ "$(PYTHON)" != "" ] ; then \
! 	    $(PYTHON) $(srcdir)/gentest.py $(srcdir) ; fi )
  
  # that one is just to make sure it is rebuilt if missing
  # but adding the dependances generate mess
! testapi.c: $(srcdir)/gentest.py
! 	-@(if [ "$(PYTHON)" != "" ] ; then \
! 	    $(PYTHON) $(srcdir)/gentest.py $(srcdir) ; fi )
! 
! testapi_SOURCES=testapi.c
! testapi_LDFLAGS = 
! testapi_DEPENDENCIES = $(DEPS)
! testapi_LDADD= $(LDADDS)
! 
! runxmlconf_SOURCES=runxmlconf.c
! runxmlconf_LDFLAGS = 
! runxmlconf_DEPENDENCIES = $(DEPS)
! runxmlconf_LDADD= $(LDADDS)
  
  #testOOM_SOURCES=testOOM.c testOOMlib.h testOOMlib.c
  #testOOM_LDFLAGS = 
  #testOOM_DEPENDENCIES = $(DEPS)
  #testOOM_LDADD= $(LDADDS)
  
! runtests:
! 	$(CHECKER) ./runtest$(EXEEXT) && $(CHECKER) ./testrecurse$(EXEEXT) &&$(CHECKER) ./testapi$(EXEEXT) && $(CHECKER) ./testchar$(EXEEXT)&& $(CHECKER) ./testdict$(EXEEXT) && $(CHECKER) ./runxmlconf$(EXEEXT)
! 	@(if [ "@PYTHON_SUBDIR@" != "" ] ; then cd python ; \
! 	    $(MAKE) MAKEFLAGS+=--silent tests ; fi)
  
  check: all runtests
  
--- 51,185 ----
  m4datadir = $(datadir)/aclocal
  m4data_DATA = libxml.m4
  
! #runtest_SOURCES=runtest.c
! #runtest_LDFLAGS = 
! #runtest_DEPENDENCIES = $(DEPS)
! #runtest_LDADD= @BASE_THREAD_LIBS@ @RDL_LIBS@ $(LDADDS)
! 
! #testrecurse_SOURCES=testrecurse.c
! #testrecurse_LDFLAGS = 
! #testrecurse_DEPENDENCIES = $(DEPS)
! #testrecurse_LDADD= @BASE_THREAD_LIBS@ @RDL_LIBS@ $(LDADDS)
! 
! #testchar_SOURCES=testchar.c
! #testchar_LDFLAGS = 
! #testchar_DEPENDENCIES = $(DEPS)
! #testchar_LDADD= @RDL_LIBS@ $(LDADDS)
! 
! #testdict_SOURCES=testdict.c
! #testdict_LDFLAGS = 
! #testdict_DEPENDENCIES = $(DEPS)
! #testdict_LDADD= @RDL_LIBS@ $(LDADDS)
! 
! #runsuite_SOURCES=runsuite.c
! #runsuite_LDFLAGS = 
! #runsuite_DEPENDENCIES = $(DEPS)
! #runsuite_LDADD= @RDL_LIBS@ $(LDADDS)
  
  xmllint_SOURCES=xmllint.c
  xmllint_LDFLAGS = 
  xmllint_DEPENDENCIES = $(DEPS)
  xmllint_LDADD=  @RDL_LIBS@ $(LDADDS)
  
! #testSAX_SOURCES=testSAX.c
! #testSAX_LDFLAGS = 
! #testSAX_DEPENDENCIES = $(DEPS)
! #testSAX_LDADD= $(LDADDS)
! 
! #testHTML_SOURCES=testHTML.c
! #testHTML_LDFLAGS = 
! #testHTML_DEPENDENCIES = $(DEPS)
! #testHTML_LDADD= $(LDADDS)
  
  xmlcatalog_SOURCES=xmlcatalog.c
  xmlcatalog_LDFLAGS = 
  xmlcatalog_DEPENDENCIES = $(DEPS)
  xmlcatalog_LDADD= @RDL_LIBS@ $(LDADDS)
  
! #testXPath_SOURCES=testXPath.c
! #testXPath_LDFLAGS = 
! #testXPath_DEPENDENCIES = $(DEPS)
! #testXPath_LDADD= $(LDADDS)
! 
! #testC14N_SOURCES=testC14N.c
! #testC14N_LDFLAGS = 
! #testC14N_DEPENDENCIES = $(DEPS)
! #testC14N_LDADD= $(LDADDS)
! 
! #testThreads_SOURCES=testThreads@THREADS_W32@.c
! #testThreads_LDFLAGS = 
! #testThreads_DEPENDENCIES = $(DEPS)
! #testThreads_LDADD= @BASE_THREAD_LIBS@  $(LDADDS)
! 
! #testURI_SOURCES=testURI.c
! #testURI_LDFLAGS = 
! #testURI_DEPENDENCIES = $(DEPS)
! #testURI_LDADD= $(LDADDS)
! 
! #testRegexp_SOURCES=testRegexp.c
! #testRegexp_LDFLAGS = 
! #testRegexp_DEPENDENCIES = $(DEPS)
! #testRegexp_LDADD= $(LDADDS)
! 
! #testAutomata_SOURCES=testAutomata.c
! #testAutomata_LDFLAGS = 
! #testAutomata_DEPENDENCIES = $(DEPS)
! #testAutomata_LDADD= $(LDADDS)
! 
! #testSchemas_SOURCES=testSchemas.c
! #testSchemas_LDFLAGS = 
! #testSchemas_DEPENDENCIES = $(DEPS)
! #testSchemas_LDADD= $(LDADDS)
! 
! #testRelax_SOURCES=testRelax.c
! #testRelax_LDFLAGS = 
! #testRelax_DEPENDENCIES = $(DEPS)
! #testRelax_LDADD= $(LDADDS)
! 
! #testReader_SOURCES=testReader.c
! #testReader_LDFLAGS = 
! #testReader_DEPENDENCIES = $(DEPS)
! #testReader_LDADD= $(LDADDS)
! 
! #testModule_SOURCES=testModule.c
! #testModule_LDFLAGS = 
! #testModule_DEPENDENCIES = $(DEPS)
! #testModule_LDADD= $(LDADDS)
! 
! #noinst_LTLIBRARIES = testdso.la
! #testdso_la_SOURCES = testdso.c
! #testdso_la_LDFLAGS = -module -rpath $(libdir)
  
  # that one forces the rebuild when "make rebuild" is run on doc/
! #rebuild_testapi:
! #	-@(if [ "$(PYTHON)" != "" ] ; then \
! #	    $(PYTHON) $(srcdir)/gentest.py $(srcdir) ; fi )
  
  # that one is just to make sure it is rebuilt if missing
  # but adding the dependances generate mess
! #testapi.c: $(srcdir)/gentest.py
! #	-@(if [ "$(PYTHON)" != "" ] ; then \
! #	    $(PYTHON) $(srcdir)/gentest.py $(srcdir) ; fi )
! 
! #testapi_SOURCES=testapi.c
! #testapi_LDFLAGS = 
! #testapi_DEPENDENCIES = $(DEPS)
! #testapi_LDADD= $(LDADDS)
! 
! #runxmlconf_SOURCES=runxmlconf.c
! #runxmlconf_LDFLAGS = 
! #runxmlconf_DEPENDENCIES = $(DEPS)
! #runxmlconf_LDADD= $(LDADDS)
  
  #testOOM_SOURCES=testOOM.c testOOMlib.h testOOMlib.c
  #testOOM_LDFLAGS = 
  #testOOM_DEPENDENCIES = $(DEPS)
  #testOOM_LDADD= $(LDADDS)
  
! #runtests:
! #	$(CHECKER) ./runtest$(EXEEXT) && $(CHECKER) ./testrecurse$(EXEEXT) &&$(CHECKER) ./testapi$(EXEEXT) && $(CHECKER) ./testchar$(EXEEXT)&& $(CHECKER) ./testdict$(EXEEXT) && $(CHECKER) ./runxmlconf$(EXEEXT)
! #	@(if [ "@PYTHON_SUBDIR@" != "" ] ; then cd python ; \
! #	    $(MAKE) MAKEFLAGS+=--silent tests ; fi)
  
  check: all runtests
  
diff -bcrN libxml2-2.7.3.org/Makefile.in libxml2-2.7.3/Makefile.in
*** libxml2-2.7.3.org/Makefile.in	2009-06-29 12:11:01.916857491 +0800
--- libxml2-2.7.3/Makefile.in	2009-06-30 00:47:21.832646154 +0800
***************
*** 35,41 ****
  POST_UNINSTALL = :
  build_triplet = @build@
  host_triplet = @host@
! noinst_PROGRAMS = testSchemas$(EXEEXT) testRelax$(EXEEXT) \
  	testSAX$(EXEEXT) testHTML$(EXEEXT) testXPath$(EXEEXT) \
  	testURI$(EXEEXT) testThreads$(EXEEXT) testC14N$(EXEEXT) \
  	testAutomata$(EXEEXT) testRegexp$(EXEEXT) testReader$(EXEEXT) \
--- 35,41 ----
  POST_UNINSTALL = :
  build_triplet = @build@
  host_triplet = @host@
! #noinst_PROGRAMS = testSchemas$(EXEEXT) testRelax$(EXEEXT) \
  	testSAX$(EXEEXT) testHTML$(EXEEXT) testXPath$(EXEEXT) \
  	testURI$(EXEEXT) testThreads$(EXEEXT) testC14N$(EXEEXT) \
  	testAutomata$(EXEEXT) testRegexp$(EXEEXT) testReader$(EXEEXT) \
***************
*** 73,79 ****
  	"$(DESTDIR)$(man3dir)" "$(DESTDIR)$(confexecdir)" \
  	"$(DESTDIR)$(m4datadir)" "$(DESTDIR)$(pkgconfigdir)"
  libLTLIBRARIES_INSTALL = $(INSTALL)
! LTLIBRARIES = $(lib_LTLIBRARIES) $(noinst_LTLIBRARIES)
  am__DEPENDENCIES_1 =
  libxml2_la_DEPENDENCIES = $(am__DEPENDENCIES_1)
  am__libxml2_la_SOURCES_DIST = SAX.c entities.c encoding.c error.c \
--- 73,79 ----
  	"$(DESTDIR)$(man3dir)" "$(DESTDIR)$(confexecdir)" \
  	"$(DESTDIR)$(m4datadir)" "$(DESTDIR)$(pkgconfigdir)"
  libLTLIBRARIES_INSTALL = $(INSTALL)
! LTLIBRARIES = $(lib_LTLIBRARIES) #$(noinst_LTLIBRARIES)
  am__DEPENDENCIES_1 =
  libxml2_la_DEPENDENCIES = $(am__DEPENDENCIES_1)
  am__libxml2_la_SOURCES_DIST = SAX.c entities.c encoding.c error.c \
***************
*** 127,133 ****
  	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
  	$(testdso_la_LDFLAGS) $(LDFLAGS) -o $@
  binPROGRAMS_INSTALL = $(INSTALL_PROGRAM)
! PROGRAMS = $(bin_PROGRAMS) $(noinst_PROGRAMS)
  am_runsuite_OBJECTS = runsuite.$(OBJEXT)
  runsuite_OBJECTS = $(am_runsuite_OBJECTS)
  am__DEPENDENCIES_2 = $(top_builddir)/libxml2.la $(am__DEPENDENCIES_1)
--- 127,133 ----
  	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
  	$(testdso_la_LDFLAGS) $(LDFLAGS) -o $@
  binPROGRAMS_INSTALL = $(INSTALL_PROGRAM)
! PROGRAMS = $(bin_PROGRAMS) #$(noinst_PROGRAMS)
  am_runsuite_OBJECTS = runsuite.$(OBJEXT)
  runsuite_OBJECTS = $(am_runsuite_OBJECTS)
  am__DEPENDENCIES_2 = $(top_builddir)/libxml2.la $(am__DEPENDENCIES_1)
***************
*** 621,627 ****
  testModule_LDFLAGS = 
  testModule_DEPENDENCIES = $(DEPS)
  testModule_LDADD = $(LDADDS)
! noinst_LTLIBRARIES = testdso.la
  testdso_la_SOURCES = testdso.c
  testdso_la_LDFLAGS = -module -rpath $(libdir)
  testapi_SOURCES = testapi.c
--- 621,627 ----
  testModule_LDFLAGS = 
  testModule_DEPENDENCIES = $(DEPS)
  testModule_LDADD = $(LDADDS)
! #noinst_LTLIBRARIES = testdso.la
  testdso_la_SOURCES = testdso.c
  testdso_la_LDFLAGS = -module -rpath $(libdir)
  testapi_SOURCES = testapi.c
***************
*** 756,764 ****
  	  rm -f "$${dir}/so_locations"; \
  	done
  
! clean-noinstLTLIBRARIES:
! 	-test -z "$(noinst_LTLIBRARIES)" || rm -f $(noinst_LTLIBRARIES)
! 	@list='$(noinst_LTLIBRARIES)'; for p in $$list; do \
  	  dir="`echo $$p | sed -e 's|/[^/]*$$||'`"; \
  	  test "$$dir" != "$$p" || dir=.; \
  	  echo "rm -f \"$${dir}/so_locations\""; \
--- 756,764 ----
  	  rm -f "$${dir}/so_locations"; \
  	done
  
! #clean-noinstLTLIBRARIES:
! #	-test -z "$(noinst_LTLIBRARIES)" || rm -f $(noinst_LTLIBRARIES)
! #	@list='$(noinst_LTLIBRARIES)'; for p in $$list; do \
  	  dir="`echo $$p | sed -e 's|/[^/]*$$||'`"; \
  	  test "$$dir" != "$$p" || dir=.; \
  	  echo "rm -f \"$${dir}/so_locations\""; \
***************
*** 797,804 ****
  	  rm -f $$p $$f ; \
  	done
  
! clean-noinstPROGRAMS:
! 	@list='$(noinst_PROGRAMS)'; for p in $$list; do \
  	  f=`echo $$p|sed 's/$(EXEEXT)$$//'`; \
  	  echo " rm -f $$p $$f"; \
  	  rm -f $$p $$f ; \
--- 797,804 ----
  	  rm -f $$p $$f ; \
  	done
  
! #clean-noinstPROGRAMS:
! #	@list='$(noinst_PROGRAMS)'; for p in $$list; do \
  	  f=`echo $$p|sed 's/$(EXEEXT)$$//'`; \
  	  echo " rm -f $$p $$f"; \
  	  rm -f $$p $$f ; \
