diff -bcrN icu.org/source/aclocal.m4 icu/source/aclocal.m4
*** icu.org/source/aclocal.m4	2009-07-05 08:52:39.607390567 +0800
--- icu/source/aclocal.m4	2009-07-09 21:30:04.478612982 +0800
***************
*** 64,69 ****
--- 64,70 ----
  *-dec-osf*) icu_cv_host_frag=mh-alpha-osf ;;
  *-*-nto*)	icu_cv_host_frag=mh-qnx ;;
  *-ncr-*)	icu_cv_host_frag=mh-mpras ;;
+ *-*-es*)        icu_cv_host_frag=mh-es ;;
  *) 		icu_cv_host_frag=mh-unknown ;;
  esac
  		]
***************
*** 123,129 ****
  AC_DEFUN(AC_CHECK_64BIT_LIBS,
  [
      BITS_REQ=nochange
!     ENABLE_64BIT_LIBS=unknown
      ## revisit this for cross-compile.
      
      AC_ARG_ENABLE(64bit-libs,
--- 124,130 ----
  AC_DEFUN(AC_CHECK_64BIT_LIBS,
  [
      BITS_REQ=nochange
!     ENABLE_64BIT_LIBS=no
      ## revisit this for cross-compile.
      
      AC_ARG_ENABLE(64bit-libs,
***************
*** 156,161 ****
--- 157,163 ----
      BITS_GOT=unknown
      
      # 'OK' here means, we can exit any further checking, everything's copa
+     DEFAULT_64BIT=no
      BITS_OK=yes
  
      # do we need to check for buildable/runnable 32 or 64 bit?
diff -bcrN icu.org/source/common/putil.c icu/source/common/putil.c
*** icu.org/source/common/putil.c	2009-07-05 08:52:39.771388018 +0800
--- icu/source/common/putil.c	2009-07-09 21:30:04.478612982 +0800
***************
*** 606,612 ****
  /* Note that U_TZNAME does *not* have to be tzname, but if it is,
     some platforms need to have it declared here. */
  
! #if defined(U_TZNAME) && (defined(U_IRIX) || defined(U_DARWIN) || defined(U_CYGWIN))
  /* RS6000 and others reject char **tzname.  */
  extern U_IMPORT char *U_TZNAME[];
  #endif
--- 606,612 ----
  /* Note that U_TZNAME does *not* have to be tzname, but if it is,
     some platforms need to have it declared here. */
  
! #if defined(U_TZNAME) && (defined(U_IRIX) || defined(U_DARWIN) || defined(U_CYGWIN) || defined(U_ES))
  /* RS6000 and others reject char **tzname.  */
  extern U_IMPORT char *U_TZNAME[];
  #endif
diff -bcrN icu.org/source/common/unicode/uconfig.h icu/source/common/unicode/uconfig.h
*** icu.org/source/common/unicode/uconfig.h	2009-07-05 08:52:39.771388018 +0800
--- icu/source/common/unicode/uconfig.h	2009-07-09 21:30:04.478612982 +0800
***************
*** 195,201 ****
   * @stable ICU 2.4
   */
  #ifndef UCONFIG_NO_FORMATTING
! #   define UCONFIG_NO_FORMATTING 0
  #endif
  
  /**
--- 195,201 ----
   * @stable ICU 2.4
   */
  #ifndef UCONFIG_NO_FORMATTING
! #   define UCONFIG_NO_FORMATTING 1
  #endif
  
  /**
diff -bcrN icu.org/source/config/mh-es icu/source/config/mh-es
*** icu.org/source/config/mh-es	1970-01-01 08:00:00.000000000 +0800
--- icu/source/config/mh-es	2009-07-09 21:30:04.478612982 +0800
***************
*** 0 ****
--- 1,78 ----
+ ## -*-makefile-*-
+ ## Linux-specific setup
+ ## Copyright (c) 1999-2008, International Business Machines Corporation and
+ ## others. All Rights Reserved.
+ 
+ ## Commands to generate dependency files
+ GEN_DEPS.c=  $(CC) -E -MM $(DEFS) $(CPPFLAGS)
+ GEN_DEPS.cc= $(CXX) -E -MM $(DEFS) $(CPPFLAGS)
+ 
+ ## Flags for position independent code
+ SHAREDLIBCFLAGS = -fPIC
+ SHAREDLIBCXXFLAGS = -fPIC
+ SHAREDLIBCPPFLAGS = -DPIC
+ 
+ ## Additional flags when building libraries and with threads
+ THREADSCPPFLAGS = -D_REENTRANT
+ LIBCPPFLAGS =
+ 
+ ## Compiler switch to embed a runtime search path
+ LD_RPATH=
+ LD_RPATH_PRE = -Wl,-rpath,
+ 
+ ## These are the library specific LDFLAGS
+ LDFLAGSICUDT=-nodefaultlibs -nostdlib
+ 
+ ## Compiler switch to embed a library name
+ # The initial tab in the next line is to prevent icu-config from reading it.
+ 	LD_SONAME = -Wl,-soname -Wl,$(notdir $(MIDDLE_SO_TARGET))
+ #SH# # We can't depend on MIDDLE_SO_TARGET being set.
+ #SH# LD_SONAME=
+ 
+ ## Shared object suffix
+ SO = so
+ ## Non-shared intermediate object suffix
+ STATIC_O = ao
+ 
+ ## Compilation rules
+ %.$(STATIC_O): $(srcdir)/%.c
+ 	$(COMPILE.c) $(STATICCPPFLAGS) $(STATICCFLAGS) -o $@ $<
+ %.o: $(srcdir)/%.c
+ 	$(COMPILE.c) $(DYNAMICCPPFLAGS) $(DYNAMICCFLAGS) -o $@ $<
+ 
+ %.$(STATIC_O): $(srcdir)/%.cpp
+ 	$(COMPILE.cc) $(STATICCPPFLAGS) $(STATICCXXFLAGS) -o $@ $<
+ %.o: $(srcdir)/%.cpp
+ 	$(COMPILE.cc) $(DYNAMICCPPFLAGS) $(DYNAMICCXXFLAGS) -o $@ $<
+ 
+ 
+ ## Dependency rules
+ %.d: $(srcdir)/%.c
+ 	@echo "generating dependency information for $<"
+ 	@$(SHELL) -ec '$(GEN_DEPS.c) $< \
+ 		| sed '\''s%\($*\)\.o[ :]*%\1.o $@ : %g'\'' > $@; \
+ 		[ -s $@ ] || rm -f $@'
+ 
+ %.d: $(srcdir)/%.cpp
+ 	@echo "generating dependency information for $<"
+ 	@$(SHELL) -ec '$(GEN_DEPS.cc) $< \
+ 		| sed '\''s%\($*\)\.o[ :]*%\1.o $@ : %g'\'' > $@; \
+ 		[ -s $@ ] || rm -f $@'
+ 
+ ## Versioned libraries rules
+ 
+ %.$(SO).$(SO_TARGET_VERSION_MAJOR): %.$(SO).$(SO_TARGET_VERSION)
+ 	$(RM) $@ && ln -s ${<F} $@
+ %.$(SO): %.$(SO).$(SO_TARGET_VERSION_MAJOR)
+ 	$(RM) $@ && ln -s ${*F}.$(SO).$(SO_TARGET_VERSION) $@
+ 
+ ##  Bind internal references
+ 
+ # LDflags that pkgdata will use
+ BIR_LDFLAGS= -Wl,-Bsymbolic
+ 
+ # Dependencies [i.e. map files] for the final library
+ BIR_DEPS=
+ 
+ ## End Linux-specific setup
+ 
diff -bcrN icu.org/source/config.sub icu/source/config.sub
*** icu.org/source/config.sub	2009-07-05 08:52:39.931388268 +0800
--- icu/source/config.sub	2009-07-09 21:30:04.478612982 +0800
***************
*** 1249,1255 ****
  	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
  	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
  	      | -chorusos* | -chorusrdb* \
! 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
  	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
  	      | -uxpv* | -beos* | -mpeix* | -udk* \
  	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
--- 1249,1255 ----
  	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
  	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
  	      | -chorusos* | -chorusrdb* \
! 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* | -es* \
  	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
  	      | -uxpv* | -beos* | -mpeix* | -udk* \
  	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
diff -bcrN icu.org/source/configure icu/source/configure
*** icu.org/source/configure	2009-07-05 08:52:39.587387584 +0800
--- icu/source/configure	2009-07-09 21:30:04.486612503 +0800
***************
*** 4706,4711 ****
--- 4706,4712 ----
  		icu_cv_host_frag=mh-linux-va
  	fi ;;
  *-*-linux*|*-pc-gnu) icu_cv_host_frag=mh-linux ;;
+ *-*-es*) icu_cv_host_frag=mh-es ;;
  *-*-cygwin|*-*-mingw32)
  	if test "$GCC" = yes; then
  		cat >conftest.$ac_ext <<_ACEOF
***************
*** 4780,4785 ****
--- 4781,4787 ----
  *-dec-osf*) icu_cv_host_frag=mh-alpha-osf ;;
  *-*-nto*)	icu_cv_host_frag=mh-qnx ;;
  *-ncr-*)	icu_cv_host_frag=mh-mpras ;;
+ *-*-es*)        icu_cv_host_frag=mh-es ;;
  *) 		icu_cv_host_frag=mh-unknown ;;
  esac
  
***************
*** 10339,10344 ****
--- 10341,10347 ----
  	*-*-beos)		platform=U_BEOS ;;
  	*-*-irix*)		platform=U_IRIX ;;
  	*-ncr-*)		platform=U_MPRAS ;;
+ 	*-*-es*)        platform=U_ES ;;
  	*) 		platform=U_UNKNOWN_PLATFORM ;;
  esac
  
diff -bcrN icu.org/source/configure.in icu/source/configure.in
*** icu.org/source/configure.in	2009-07-05 08:52:39.587387584 +0800
--- icu/source/configure.in	2009-07-09 21:30:30.342613280 +0800
***************
*** 107,112 ****
--- 107,113 ----
  AC_MSG_CHECKING(checking for executable suffix)
  case "${host}" in
      *-*-cygwin*|*-*-mingw*)  EXEEXT=.exe ;;
+     *-*-es*)  EXEEXT=.elf ;;
      *)      EXEEXT="" ;;
  esac
  ac_exeext=$EXEEXT
***************
*** 460,466 ****
  AC_MSG_CHECKING([for genccode assembly])
  
  # Check to see if genccode can generate simple assembly.
! GENCCODE_ASSEMBLY=
  case "${host}" in
  i*86-*-linux*|x86_64-*-linux*|powerpc*-*-linux*|i*86-*-*bsd*|i*86-pc-gnu)
      if test "$GCC" = yes; then
--- 461,467 ----
  AC_MSG_CHECKING([for genccode assembly])
  
  # Check to see if genccode can generate simple assembly.
! GENCCODE_ASSEMBLY="-a gcc"
  case "${host}" in
  i*86-*-linux*|x86_64-*-linux*|powerpc*-*-linux*|i*86-*-*bsd*|i*86-pc-gnu)
      if test "$GCC" = yes; then
***************
*** 625,634 ****
      ], [f();])],[ac_cv_namespace_ok=yes],[ac_cv_namespace_ok=no])] )
  AC_MSG_RESULT($ac_cv_namespace_ok)
  U_HAVE_NAMESPACE=1
! if test $ac_cv_namespace_ok = no
! then
!     U_HAVE_NAMESPACE=0
! fi
  AC_SUBST(U_HAVE_NAMESPACE)
  
  AC_MSG_CHECKING([for properly overriding new and delete])
--- 626,635 ----
      ], [f();])],[ac_cv_namespace_ok=yes],[ac_cv_namespace_ok=no])] )
  AC_MSG_RESULT($ac_cv_namespace_ok)
  U_HAVE_NAMESPACE=1
! #if test $ac_cv_namespace_ok = no
! #then
! #    U_HAVE_NAMESPACE=0
! #fi
  AC_SUBST(U_HAVE_NAMESPACE)
  
  AC_MSG_CHECKING([for properly overriding new and delete])
***************
*** 732,738 ****
  #endif
  #include <time.h>
  ], [[timezone = 1;]])],[ac_cv_var_timezone=yes],[ac_cv_var_timezone=no])])
! U_HAVE_TIMEZONE=0
  if test $ac_cv_var_timezone = yes; then
    U_TIMEZONE=timezone
    U_HAVE_TIMEZONE=1
--- 733,739 ----
  #endif
  #include <time.h>
  ], [[timezone = 1;]])],[ac_cv_var_timezone=yes],[ac_cv_var_timezone=no])])
! U_HAVE_TIMEZONE=1
  if test $ac_cv_var_timezone = yes; then
    U_TIMEZONE=timezone
    U_HAVE_TIMEZONE=1
***************
*** 1087,1092 ****
--- 1088,1094 ----
  	*-*-beos)		platform=U_BEOS ;;
  	*-*-irix*)		platform=U_IRIX ;;
  	*-ncr-*)		platform=U_MPRAS ;;
+ 	*-*-es*)        platform=U_ES ;;
  	*) 		platform=U_UNKNOWN_PLATFORM ;;
  esac
  AC_SUBST(ICUDATA_CHAR)
diff -bcrN icu.org/source/data/Makefile.in icu/source/data/Makefile.in
*** icu.org/source/data/Makefile.in	2009-07-05 08:52:39.931388268 +0800
--- icu/source/data/Makefile.in	2009-07-09 21:30:52.206612966 +0800
***************
*** 462,468 ****
  #          The $(word n, ...) selects the nth word from the following stuff.
  #          There must be a nicer way to do this.
  
! $(CFU_FILES): $(ALL_CFU_SOURCE) $(TOOLBINDIR)/gencfu$(EXEEXT) $(DAT_FILES)
  	$(INVOKE) echo ALL_CFU_SOURCE: $(ALL_CFU_SOURCE)
  	$(INVOKE) echo CFU_FILES: $(CFU_FILES)
  	$(INVOKE) echo CFU_FILES_SHORT: $(CFU_FILES_SHORT)
--- 462,468 ----
  #          The $(word n, ...) selects the nth word from the following stuff.
  #          There must be a nicer way to do this.
  
! $(CFU_FILES): $(ALL_CFU_SOURCE) $(TOOLBINDIR)/gencfu$(TOOLEXEEXT) $(DAT_FILES)
  	$(INVOKE) echo ALL_CFU_SOURCE: $(ALL_CFU_SOURCE)
  	$(INVOKE) echo CFU_FILES: $(CFU_FILES)
  	$(INVOKE) echo CFU_FILES_SHORT: $(CFU_FILES_SHORT)
diff -bcrN icu.org/source/tools/gentest/gentest.c icu/source/tools/gentest/gentest.c
*** icu.org/source/tools/gentest/gentest.c	2009-07-05 08:52:39.699388169 +0800
--- icu/source/tools/gentest/gentest.c	2009-07-09 21:30:04.486612503 +0800
***************
*** 196,210 ****
              fprintf(out, 
                  "           ");
              switch(t) {
!             case UDBG_UCalendarDateFields:
              case UDBG_UCalendarMonths:
!                 /* Temporary workaround for IS_LEAP_MOTH #6051 */
                  if (t == UDBG_UCalendarDateFields && i == 22) {
!                     fprintf(out, "com.ibm.icu.util.ChineseCalendar.%s, /* %d */", udbg_enumName((UDebugEnumType)t,i), i);
                  } else
!                 fprintf(out, "com.ibm.icu.util.Calendar.%s, /* %d */", udbg_enumName((UDebugEnumType)t,i), i);
                  break;
!             case UDBG_UDebugEnumType:
              default:
                  fprintf(out,"%d, /* %s */", i, udbg_enumName((UDebugEnumType)t,i));
              }
--- 196,210 ----
              fprintf(out, 
                  "           ");
              switch(t) {
!             /* case UDBG_UCalendarDateFields: 
              case UDBG_UCalendarMonths:
!                  Temporary workaround for IS_LEAP_MOTH #6051 
                  if (t == UDBG_UCalendarDateFields && i == 22) {
!                     fprintf(out, "com.ibm.icu.util.ChineseCalendar.%s,  %d ", udbg_enumName((UDebugEnumType)t,i), i);
                  } else
!                 fprintf(out, "com.ibm.icu.util.Calendar.%s,  %d ", udbg_enumName((UDebugEnumType)t,i), i);
                  break;
!             case UDBG_UDebugEnumType:*/
              default:
                  fprintf(out,"%d, /* %s */", i, udbg_enumName((UDebugEnumType)t,i));
              }
