diff -bcrN icu.org/source/aclocal.m4 icu/source/aclocal.m4
*** icu.org/source/aclocal.m4	2009-06-26 13:40:06.400655176 +0800
--- icu/source/aclocal.m4	2009-06-26 13:40:08.860654156 +0800
***************
*** 64,69 ****
--- 64,70 ----
  *-dec-osf*) icu_cv_host_frag=mh-alpha-osf ;;
  *-*-nto*)	icu_cv_host_frag=mh-qnx ;;
  *-ncr-*)	icu_cv_host_frag=mh-mpras ;;
+ *-*-es*)        icu_cv_host_frag=mh-es ;;
  *) 		icu_cv_host_frag=mh-unknown ;;
  esac
  		]
diff -bcrN icu.org/source/config/mh-es icu/source/config/mh-es
*** icu.org/source/config/mh-es	1970-01-01 08:00:00.000000000 +0800
--- icu/source/config/mh-es	2009-06-26 13:40:08.860654156 +0800
***************
*** 0 ****
--- 1,78 ----
+ ## -*-makefile-*-
+ ## Linux-specific setup
+ ## Copyright (c) 1999-2008, International Business Machines Corporation and
+ ## others. All Rights Reserved.
+ 
+ ## Commands to generate dependency files
+ GEN_DEPS.c=  $(CC) -E -MM $(DEFS) $(CPPFLAGS)
+ GEN_DEPS.cc= $(CXX) -E -MM $(DEFS) $(CPPFLAGS)
+ 
+ ## Flags for position independent code
+ SHAREDLIBCFLAGS = -fPIC
+ SHAREDLIBCXXFLAGS = -fPIC
+ SHAREDLIBCPPFLAGS = -DPIC
+ 
+ ## Additional flags when building libraries and with threads
+ THREADSCPPFLAGS = -D_REENTRANT
+ LIBCPPFLAGS =
+ 
+ ## Compiler switch to embed a runtime search path
+ LD_RPATH=
+ LD_RPATH_PRE = -Wl,-rpath,
+ 
+ ## These are the library specific LDFLAGS
+ LDFLAGSICUDT=-nodefaultlibs -nostdlib
+ 
+ ## Compiler switch to embed a library name
+ # The initial tab in the next line is to prevent icu-config from reading it.
+ 	LD_SONAME = -Wl,-soname -Wl,$(notdir $(MIDDLE_SO_TARGET))
+ #SH# # We can't depend on MIDDLE_SO_TARGET being set.
+ #SH# LD_SONAME=
+ 
+ ## Shared object suffix
+ SO = so
+ ## Non-shared intermediate object suffix
+ STATIC_O = ao
+ 
+ ## Compilation rules
+ %.$(STATIC_O): $(srcdir)/%.c
+ 	$(COMPILE.c) $(STATICCPPFLAGS) $(STATICCFLAGS) -o $@ $<
+ %.o: $(srcdir)/%.c
+ 	$(COMPILE.c) $(DYNAMICCPPFLAGS) $(DYNAMICCFLAGS) -o $@ $<
+ 
+ %.$(STATIC_O): $(srcdir)/%.cpp
+ 	$(COMPILE.cc) $(STATICCPPFLAGS) $(STATICCXXFLAGS) -o $@ $<
+ %.o: $(srcdir)/%.cpp
+ 	$(COMPILE.cc) $(DYNAMICCPPFLAGS) $(DYNAMICCXXFLAGS) -o $@ $<
+ 
+ 
+ ## Dependency rules
+ %.d: $(srcdir)/%.c
+ 	@echo "generating dependency information for $<"
+ 	@$(SHELL) -ec '$(GEN_DEPS.c) $< \
+ 		| sed '\''s%\($*\)\.o[ :]*%\1.o $@ : %g'\'' > $@; \
+ 		[ -s $@ ] || rm -f $@'
+ 
+ %.d: $(srcdir)/%.cpp
+ 	@echo "generating dependency information for $<"
+ 	@$(SHELL) -ec '$(GEN_DEPS.cc) $< \
+ 		| sed '\''s%\($*\)\.o[ :]*%\1.o $@ : %g'\'' > $@; \
+ 		[ -s $@ ] || rm -f $@'
+ 
+ ## Versioned libraries rules
+ 
+ %.$(SO).$(SO_TARGET_VERSION_MAJOR): %.$(SO).$(SO_TARGET_VERSION)
+ 	$(RM) $@ && ln -s ${<F} $@
+ %.$(SO): %.$(SO).$(SO_TARGET_VERSION_MAJOR)
+ 	$(RM) $@ && ln -s ${*F}.$(SO).$(SO_TARGET_VERSION) $@
+ 
+ ##  Bind internal references
+ 
+ # LDflags that pkgdata will use
+ BIR_LDFLAGS= -Wl,-Bsymbolic
+ 
+ # Dependencies [i.e. map files] for the final library
+ BIR_DEPS=
+ 
+ ## End Linux-specific setup
+ 
diff -bcrN icu.org/source/config.sub icu/source/config.sub
*** icu.org/source/config.sub	2009-06-26 13:40:06.944656965 +0800
--- icu/source/config.sub	2009-06-26 13:40:08.860654156 +0800
***************
*** 1249,1255 ****
  	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
  	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
  	      | -chorusos* | -chorusrdb* \
! 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
  	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
  	      | -uxpv* | -beos* | -mpeix* | -udk* \
  	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
--- 1249,1255 ----
  	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
  	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
  	      | -chorusos* | -chorusrdb* \
! 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* | -es* \
  	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
  	      | -uxpv* | -beos* | -mpeix* | -udk* \
  	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
diff -bcrN icu.org/source/configure icu/source/configure
*** icu.org/source/configure	2009-06-26 13:40:06.372654024 +0800
--- icu/source/configure	2009-06-26 13:48:15.904652076 +0800
***************
*** 3990,3996 ****
  
  
      BITS_REQ=nochange
!     ENABLE_64BIT_LIBS=unknown
      ## revisit this for cross-compile.
  
      # Check whether --enable-64bit-libs was given.
--- 3990,3996 ----
  
  
      BITS_REQ=nochange
!     ENABLE_64BIT_LIBS=no
      ## revisit this for cross-compile.
  
      # Check whether --enable-64bit-libs was given.
***************
*** 4071,4077 ****
  rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
  fi
  
! 
      BITS_GOT=unknown
  
      # 'OK' here means, we can exit any further checking, everything's copa
--- 4071,4077 ----
  rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext conftest.$ac_objext conftest.$ac_ext
  fi
  
!     DEFAULT_64BIT=no
      BITS_GOT=unknown
  
      # 'OK' here means, we can exit any further checking, everything's copa
***************
*** 10339,10344 ****
--- 10339,10345 ----
  	*-*-beos)		platform=U_BEOS ;;
  	*-*-irix*)		platform=U_IRIX ;;
  	*-ncr-*)		platform=U_MPRAS ;;
+ 	*-*-es*)        platform=U_ES ;;
  	*) 		platform=U_UNKNOWN_PLATFORM ;;
  esac
  
diff -bcrN icu.org/source/configure.in icu/source/configure.in
*** icu.org/source/configure.in	2009-06-26 13:40:06.372654024 +0800
--- icu/source/configure.in	2009-06-26 14:27:12.484653902 +0800
***************
*** 460,466 ****
  AC_MSG_CHECKING([for genccode assembly])
  
  # Check to see if genccode can generate simple assembly.
! GENCCODE_ASSEMBLY=
  case "${host}" in
  i*86-*-linux*|x86_64-*-linux*|powerpc*-*-linux*|i*86-*-*bsd*|i*86-pc-gnu)
      if test "$GCC" = yes; then
--- 460,466 ----
  AC_MSG_CHECKING([for genccode assembly])
  
  # Check to see if genccode can generate simple assembly.
! GENCCODE_ASSEMBLY="-a gcc"
  case "${host}" in
  i*86-*-linux*|x86_64-*-linux*|powerpc*-*-linux*|i*86-*-*bsd*|i*86-pc-gnu)
      if test "$GCC" = yes; then
***************
*** 625,634 ****
      ], [f();])],[ac_cv_namespace_ok=yes],[ac_cv_namespace_ok=no])] )
  AC_MSG_RESULT($ac_cv_namespace_ok)
  U_HAVE_NAMESPACE=1
! if test $ac_cv_namespace_ok = no
! then
!     U_HAVE_NAMESPACE=0
! fi
  AC_SUBST(U_HAVE_NAMESPACE)
  
  AC_MSG_CHECKING([for properly overriding new and delete])
--- 625,634 ----
      ], [f();])],[ac_cv_namespace_ok=yes],[ac_cv_namespace_ok=no])] )
  AC_MSG_RESULT($ac_cv_namespace_ok)
  U_HAVE_NAMESPACE=1
! #if test $ac_cv_namespace_ok = no
! #then
! #    U_HAVE_NAMESPACE=0
! #fi
  AC_SUBST(U_HAVE_NAMESPACE)
  
  AC_MSG_CHECKING([for properly overriding new and delete])
***************
*** 1087,1092 ****
--- 1087,1093 ----
  	*-*-beos)		platform=U_BEOS ;;
  	*-*-irix*)		platform=U_IRIX ;;
  	*-ncr-*)		platform=U_MPRAS ;;
+ 	*-*-es*)        platform=U_ES ;;
  	*) 		platform=U_UNKNOWN_PLATFORM ;;
  esac
  AC_SUBST(ICUDATA_CHAR)
