diff -bcrN pcre-7.8.org/config.sub pcre-7.8/config.sub
*** pcre-7.8.org/config.sub	2008-09-06 00:58:04.000000000 +0900
--- pcre-7.8/config.sub	2008-09-08 16:18:37.000000000 +0900
***************
*** 1276,1282 ****
  	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
  	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
  	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
! 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -irx*)
  	# Remember, each alternative MUST END IN *, to match a version number.
  		;;
  	-qnx*)
--- 1276,1282 ----
  	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
  	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
  	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
! 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -irx* | -es*)
  	# Remember, each alternative MUST END IN *, to match a version number.
  		;;
  	-qnx*)
diff -bcrN pcre-7.8.org/pcretest.c pcre-7.8/pcretest.c
*** pcre-7.8.org/pcretest.c	2008-08-26 03:27:26.000000000 +0900
--- pcre-7.8/pcretest.c	2008-09-08 16:22:21.000000000 +0900
***************
*** 901,907 ****
        ((stack_size = get_value((unsigned char *)argv[op+1], &endptr)),
          *endptr == 0))
      {
! #if defined(_WIN32) || defined(WIN32)
      printf("PCRE: -S not supported on this OS\n");
      exit(1);
  #else
--- 901,907 ----
        ((stack_size = get_value((unsigned char *)argv[op+1], &endptr)),
          *endptr == 0))
      {
! #if defined(_WIN32) || defined(WIN32) || defined(__es__)
      printf("PCRE: -S not supported on this OS\n");
      exit(1);
  #else
