diff -bcrN freetype-2.3.4.org/builds/unix/config.sub freetype-2.3.4/builds/unix/config.sub
*** freetype-2.3.4.org/builds/unix/config.sub	2007-04-10 13:12:50.000000000 +0900
--- freetype-2.3.4/builds/unix/config.sub	2007-06-26 18:04:12.000000000 +0900
***************
*** 1213,1219 ****
  	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
  	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
  	      | -chorusos* | -chorusrdb* \
! 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
  	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
  	      | -uxpv* | -beos* | -mpeix* | -udk* \
  	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
--- 1213,1219 ----
  	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
  	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
  	      | -chorusos* | -chorusrdb* \
! 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* | -es* \
  	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
  	      | -uxpv* | -beos* | -mpeix* | -udk* \
  	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
