diff -bcrN trunk.org/JavaScriptCore/runtime/DateMath.h trunk/JavaScriptCore/runtime/DateMath.h
*** trunk.org/JavaScriptCore/runtime/DateMath.h	2009-07-26 17:54:13.015958617 +0800
--- trunk/JavaScriptCore/runtime/DateMath.h	2009-07-26 18:30:56.144858710 +0800
***************
*** 113,119 ****
          , year(inTm.tm_year)
          , isDST(inTm.tm_isdst)
      {
! #if !PLATFORM(WIN_OS) && !PLATFORM(SOLARIS) && !COMPILER(RVCT)
          utcOffset = static_cast<int>(inTm.tm_gmtoff);
  
          int inZoneSize = strlen(inTm.tm_zone) + 1;
--- 113,119 ----
          , year(inTm.tm_year)
          , isDST(inTm.tm_isdst)
      {
! #if !PLATFORM(WIN_OS) && !PLATFORM(SOLARIS) && !PLATFORM(ES) && !COMPILER(RVCT)
          utcOffset = static_cast<int>(inTm.tm_gmtoff);
  
          int inZoneSize = strlen(inTm.tm_zone) + 1;
***************
*** 140,146 ****
          ret.tm_year  =  year;
          ret.tm_isdst =  isDST;
  
! #if !PLATFORM(WIN_OS) && !PLATFORM(SOLARIS) && !COMPILER(RVCT)
          ret.tm_gmtoff = static_cast<long>(utcOffset);
          ret.tm_zone = timeZone;
  #endif
--- 140,146 ----
          ret.tm_year  =  year;
          ret.tm_isdst =  isDST;
  
! #if !PLATFORM(WIN_OS) && !PLATFORM(SOLARIS) && !PLATFORM(ES) && !COMPILER(RVCT)
          ret.tm_gmtoff = static_cast<long>(utcOffset);
          ret.tm_zone = timeZone;
  #endif
diff -bcrN trunk.org/JavaScriptCore/wtf/Platform.h trunk/JavaScriptCore/wtf/Platform.h
*** trunk.org/JavaScriptCore/wtf/Platform.h	2009-07-26 17:54:13.704713755 +0800
--- trunk/JavaScriptCore/wtf/Platform.h	2009-07-26 18:27:00.295956128 +0800
***************
*** 341,348 ****
  #endif
  #endif
  
  #if !defined(HAVE_ACCESSIBILITY)
! #if PLATFORM(MAC) || PLATFORM(WIN) || PLATFORM(GTK) || PLATFORM(CHROMIUM)
  #define HAVE_ACCESSIBILITY 1
  #endif
  #endif /* !defined(HAVE_ACCESSIBILITY) */
--- 341,352 ----
  #endif
  #endif
  
+ #if PLATFORM(ES)
+ #define WTF_USE_PTHREADS 1
+ #endif
+ 
  #if !defined(HAVE_ACCESSIBILITY)
! #if PLATFORM(MAC) || PLATFORM(WIN) || PLATFORM(GTK) || PLATFORM(CHROMIUM) || PLATFORM(ES)
  #define HAVE_ACCESSIBILITY 1
  #endif
  #endif /* !defined(HAVE_ACCESSIBILITY) */
***************
*** 395,401 ****
  #define HAVE_LANGINFO_H 1
  #define HAVE_MMAP 1
  #define HAVE_SBRK 1
! #define HAVE_STRINGS_H 1
  #define HAVE_SYS_PARAM_H 1
  #define HAVE_SYS_TIME_H 1
  
--- 399,405 ----
  #define HAVE_LANGINFO_H 1
  #define HAVE_MMAP 1
  #define HAVE_SBRK 1
! #define HAVE_STRINGS_H 0
  #define HAVE_SYS_PARAM_H 1
  #define HAVE_SYS_TIME_H 1
  
diff -bcrN trunk.org/WebCore/config.h trunk/WebCore/config.h
*** trunk.org/WebCore/config.h	2009-07-26 17:56:17.279147742 +0800
--- trunk/WebCore/config.h	2009-07-26 18:22:04.524708017 +0800
***************
*** 162,167 ****
--- 162,171 ----
  /* Using V8 implies not using JSC and vice versa */
  #define WTF_USE_JSC !WTF_USE_V8
  
+ #if PLATFORM(ES)
+ #undef WTF_USE_JSC
+ #endif
+ 
  #if PLATFORM(CG)
  #ifndef CGFLOAT_DEFINED
  #ifdef __LP64__
diff -bcrN trunk.org/WebCore/dom/XMLTokenizerLibxml2.cpp trunk/WebCore/dom/XMLTokenizerLibxml2.cpp
*** trunk.org/WebCore/dom/XMLTokenizerLibxml2.cpp	2009-07-26 17:55:22.014712376 +0800
--- trunk/WebCore/dom/XMLTokenizerLibxml2.cpp	2009-07-26 18:22:04.524708017 +0800
***************
*** 49,54 ****
--- 49,55 ----
  #include "ScriptElement.h"
  #include "ScriptSourceCode.h"
  #include "ScriptValue.h"
+ #include "SecurityOrigin.h"
  #include "TextResourceDecoder.h"
  #include "XMLTokenizerScope.h"
  #include <libxml/parser.h>
diff -bcrN trunk.org/WebCore/loader/CachedResource.cpp trunk/WebCore/loader/CachedResource.cpp
*** trunk.org/WebCore/loader/CachedResource.cpp	2009-07-26 17:56:14.185956883 +0800
--- trunk/WebCore/loader/CachedResource.cpp	2009-07-26 18:22:04.524708017 +0800
***************
*** 35,40 ****
--- 35,41 ----
  #include <wtf/CurrentTime.h>
  #include <wtf/MathExtras.h>
  #include <wtf/RefCountedLeakCounter.h>
+ #include <wtf/StdLibExtras.h>
  #include <wtf/Vector.h>
  
  using namespace WTF;
diff -bcrN trunk.org/WebCore/page/DragController.cpp trunk/WebCore/page/DragController.cpp
*** trunk.org/WebCore/page/DragController.cpp	2009-07-26 17:55:31.854706818 +0800
--- trunk/WebCore/page/DragController.cpp	2009-07-26 18:22:04.524708017 +0800
***************
*** 55,60 ****
--- 55,61 ----
  #include "RenderImage.h"
  #include "ReplaceSelectionCommand.h"
  #include "ResourceRequest.h"
+ #include "SecurityOrigin.h"
  #include "SelectionController.h"
  #include "Settings.h"
  #include "Text.h"
diff -bcrN trunk.org/WebCore/page/Page.cpp trunk/WebCore/page/Page.cpp
*** trunk.org/WebCore/page/Page.cpp	2009-07-26 17:55:31.834707001 +0800
--- trunk/WebCore/page/Page.cpp	2009-07-26 18:22:04.524708017 +0800
***************
*** 117,123 ****
--- 117,125 ----
      , m_didLoadUserStyleSheet(false)
      , m_userStyleSheetModificationTime(0)
      , m_group(0)
+ #if ENABLE(JAVASCRIPT_DEBUGGER)
      , m_debugger(0)
+ #endif
      , m_customHTMLTokenizerTimeDelay(-1)
      , m_customHTMLTokenizerChunkSize(-1)
  {
***************
*** 527,532 ****
--- 529,535 ----
      }
  }
  
+ #if ENABLE(JAVASCRIPT_DEBUGGER)
  void Page::setDebuggerForAllPages(JSC::Debugger* debugger)
  {
      ASSERT(allPages);
***************
*** 546,551 ****
--- 549,555 ----
      for (Frame* frame = m_mainFrame.get(); frame; frame = frame->tree()->traverseNext())
          frame->script()->attachDebugger(m_debugger);
  }
+ #endif // ENABLE(JAVASCRIPT_DEBUGGER)
  
  #if ENABLE(DOM_STORAGE)
  SessionStorage* Page::sessionStorage(bool optionalCreate)
diff -bcrN trunk.org/WebCore/page/Page.h trunk/WebCore/page/Page.h
*** trunk.org/WebCore/page/Page.h	2009-07-26 17:55:31.854706818 +0800
--- trunk/WebCore/page/Page.h	2009-07-26 18:22:04.524708017 +0800
***************
*** 160,168 ****
--- 160,170 ----
          void userStyleSheetLocationChanged();
          const String& userStyleSheet() const;
  
+ #if ENABLE(JAVASCRIPT_DEBUGGER)
          static void setDebuggerForAllPages(JSC::Debugger*);
          void setDebugger(JSC::Debugger*);
          JSC::Debugger* debugger() const { return m_debugger; }
+ #endif // ENABLE(JAVASCRIPT_DEBUGGER)
  
  #if PLATFORM(WIN) || (PLATFORM(WX) && PLATFORM(WIN_OS)) || (PLATFORM(QT) && defined(Q_WS_WIN))
          // The global DLL or application instance used for all windows.
***************
*** 242,248 ****
--- 244,252 ----
          OwnPtr<PageGroup> m_singlePageGroup;
          PageGroup* m_group;
  
+ #if ENABLE(JAVASCRIPT_DEBUGGER)
          JSC::Debugger* m_debugger;
+ #endif
  
          double m_customHTMLTokenizerTimeDelay;
          int m_customHTMLTokenizerChunkSize;
diff -bcrN trunk.org/WebCore/platform/Cursor.h trunk/WebCore/platform/Cursor.h
*** trunk.org/WebCore/platform/Cursor.h	2009-07-26 17:55:49.134706966 +0800
--- trunk/WebCore/platform/Cursor.h	2009-07-26 18:22:04.524708017 +0800
***************
*** 40,45 ****
--- 40,47 ----
  #include <QCursor>
  #elif PLATFORM(CHROMIUM)
  #include "PlatformCursor.h"
+ #elif PLATFORM(ES)
+ #include "PlatformCursor.h"
  #endif
  
  #if PLATFORM(MAC)
***************
*** 86,91 ****
--- 88,96 ----
  #elif PLATFORM(CHROMIUM)
      // See PlatformCursor.h
      typedef void* PlatformCursorHandle;
+ #elif PLATFORM(ES)
+     // See PlatformCursor.h
+     typedef void* PlatformCursorHandle;
  #else
      typedef void* PlatformCursor;
      typedef void* PlatformCursorHandle;
diff -bcrN trunk.org/WebCore/platform/DragData.h trunk/WebCore/platform/DragData.h
*** trunk.org/WebCore/platform/DragData.h	2009-07-26 17:55:49.144706249 +0800
--- trunk/WebCore/platform/DragData.h	2009-07-26 18:22:04.524708017 +0800
***************
*** 56,61 ****
--- 56,63 ----
  typedef void* DragDataRef;
  #elif PLATFORM(CHROMIUM)
  #include "DragDataRef.h"
+ #elif PLATFORM(ES)
+ #include "DragDataRef.h"
  #endif
  
  
diff -bcrN trunk.org/WebCore/platform/DragImage.h trunk/WebCore/platform/DragImage.h
*** trunk.org/WebCore/platform/DragImage.h	2009-07-26 17:55:49.174707540 +0800
--- trunk/WebCore/platform/DragImage.h	2009-07-26 18:22:04.524708017 +0800
***************
*** 46,51 ****
--- 46,53 ----
  class wxDragImage;
  #elif PLATFORM(CHROMIUM)
  #include "DragImageRef.h"
+ #elif PLATFORM(ES)
+ #include "DragImageRef.h"
  #endif
  
  //We need to #define YOffset as it needs to be shared with WebKit
diff -bcrN trunk.org/WebCore/platform/graphics/cairo/ImageBufferCairo.cpp trunk/WebCore/platform/graphics/cairo/ImageBufferCairo.cpp
*** trunk.org/WebCore/platform/graphics/cairo/ImageBufferCairo.cpp	2009-07-26 17:55:45.944706293 +0800
--- trunk/WebCore/platform/graphics/cairo/ImageBufferCairo.cpp	2009-07-26 22:57:43.549356004 +0800
***************
*** 212,218 ****
  
      Vector<char> in;
      // Only PNG output is supported for now.
!     cairo_surface_write_to_png_stream(image, writeFunction, &in);
  
      Vector<char> out;
      base64Encode(in, out);
--- 212,218 ----
  
      Vector<char> in;
      // Only PNG output is supported for now.
!     //cairo_surface_write_to_png_stream(image, writeFunction, &in);
  
      Vector<char> out;
      base64Encode(in, out);
diff -bcrN trunk.org/WebCore/platform/graphics/gtk/FontPlatformDataGtk.cpp trunk/WebCore/platform/graphics/gtk/FontPlatformDataGtk.cpp
*** trunk.org/WebCore/platform/graphics/gtk/FontPlatformDataGtk.cpp	2009-07-26 17:55:43.454706292 +0800
--- trunk/WebCore/platform/graphics/gtk/FontPlatformDataGtk.cpp	2009-07-26 18:22:04.524708017 +0800
***************
*** 32,38 ****
--- 32,40 ----
  #include <cairo-ft.h>
  #include <cairo.h>
  #include <fontconfig/fcfreetype.h>
+ #if !PLATFORM(ES)
  #include <gtk/gtk.h>
+ #endif
  
  namespace WebCore {
  
***************
*** 109,116 ****
--- 111,120 ----
      cairo_matrix_init_scale(&fontMatrix, fontDescription.computedPixelSize(), fontDescription.computedPixelSize());
      cairo_matrix_init_identity(&ctm);
  
+ #if !PLATFORM(ES)
      if (GdkScreen* screen = gdk_screen_get_default())
          options = gdk_screen_get_font_options(screen);
+ #endif
  
      // gdk_screen_get_font_options() returns NULL if no default options are
      // set, so we always have to check.
***************
*** 149,156 ****
--- 153,162 ----
      static const cairo_font_options_t* defaultOptions = cairo_font_options_create();
      const cairo_font_options_t* options = NULL;
  
+ #if !PLATFORM(ES)
      if (GdkScreen* screen = gdk_screen_get_default())
          options = gdk_screen_get_font_options(screen);
+ #endif
  
      // gdk_screen_get_font_options() returns NULL if no default options are
      // set, so we always have to check.
diff -bcrN trunk.org/WebCore/platform/network/ResourceResponseBase.cpp trunk/WebCore/platform/network/ResourceResponseBase.cpp
*** trunk.org/WebCore/platform/network/ResourceResponseBase.cpp	2009-07-26 17:55:47.134708034 +0800
--- trunk/WebCore/platform/network/ResourceResponseBase.cpp	2009-07-26 18:22:04.524708017 +0800
***************
*** 324,329 ****
--- 324,330 ----
  
  static double parseDateValueInHeader(const HTTPHeaderMap& headers, const AtomicString& headerName)
  {
+ #if USE(JSC)
      String headerValue = headers.get(headerName);
      if (headerValue.isEmpty())
          return std::numeric_limits<double>::quiet_NaN(); 
***************
*** 331,340 ****
      // Sun, 06 Nov 1994 08:49:37 GMT  ; RFC 822, updated by RFC 1123
      // Sunday, 06-Nov-94 08:49:37 GMT ; RFC 850, obsoleted by RFC 1036
      // Sun Nov  6 08:49:37 1994       ; ANSI C's asctime() format
!     double dateInMilliseconds = JSC::parseDate(headerValue);
      if (!isfinite(dateInMilliseconds))
          return std::numeric_limits<double>::quiet_NaN();
      return dateInMilliseconds / 1000;
  }
  
  double ResourceResponseBase::date() const
--- 332,344 ----
      // Sun, 06 Nov 1994 08:49:37 GMT  ; RFC 822, updated by RFC 1123
      // Sunday, 06-Nov-94 08:49:37 GMT ; RFC 850, obsoleted by RFC 1036
      // Sun Nov  6 08:49:37 1994       ; ANSI C's asctime() format
!     double dateInMilliseconds = JSC::parseDate(JSC::UString(headerValue.characters(), headerValue.length()));
      if (!isfinite(dateInMilliseconds))
          return std::numeric_limits<double>::quiet_NaN();
      return dateInMilliseconds / 1000;
+ #else
+     return std::numeric_limits<double>::quiet_NaN();
+ #endif
  }
  
  double ResourceResponseBase::date() const
diff -bcrN trunk.org/WebCore/platform/Widget.h trunk/WebCore/platform/Widget.h
*** trunk.org/WebCore/platform/Widget.h	2009-07-26 17:55:49.184708769 +0800
--- trunk/WebCore/platform/Widget.h	2009-07-26 18:22:04.524708017 +0800
***************
*** 69,74 ****
--- 69,78 ----
  #include "PlatformWidget.h"
  #endif
  
+ #if PLATFORM(ES)
+ #include "PlatformWidget.h"
+ #endif
+ 
  #include "IntPoint.h"
  #include "IntRect.h"
  #include "IntSize.h"
